modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/MockedStructuresTest.java

```diff
@@ -275,6 +285,58 @@ private void mockMetastore() throws Exception {
         when(cursorMocked.iterator()).thenReturn(itMock);
     }
 
+    @Test
+    public void testCreateZone() {
+        String mtdName = getCurrentMethodName();
+
+        String query = String.format("CREATE ZONE %s", mtdName);
+
+        // Create new distribution zone.
+        readFirst(queryProc.queryAsync("PUBLIC", query));
+
+        // Create distribution zone with existing name.
+        IgniteException ex = assertThrows(IgniteException.class, () -> readFirst(queryProc.queryAsync("PUBLIC", query)));
+        assertTrue(hasCause(ex, DistributionZoneAlreadyExistsException.class, null));
+
+        // Check ifNotExists flag.
+        readFirst(queryProc.queryAsync("PUBLIC", String.format("CREATE ZONE IF NOT EXISTS %s", mtdName)));
+    }
+
+    @Test
+    public void testCreateZoneOptions() {
+        String mtdName = getCurrentMethodName();
+
+        // Check for conflicting options.
+        String qry0 = String.format("CREATE ZONE %s WITH DATA_NODES_AUTO_ADJUST=10, DATA_NODES_AUTO_ADJUST_SCALE_UP=5", mtdName);
```

Comment:
what if some options will be presented several times? I believe we need such a test

Reply:
We already have such a test - `DdlSqlToCommandConverterTest#zoneCreateDuplicateOptions`